addons/cargo/functions/fnc_unloadItem.sqf

```diff
@@ -1,57 +1,74 @@
  *
  * Public: Yes
  */
 
-params ["_item", "_vehicle", ["_unloader", objNull]];
+params [["_item", "", [objNull, ""]], ["_vehicle", objNull, [objNull]], ["_unloader", objNull, [objNull]]];
 TRACE_3("params",_item,_vehicle,_unloader);
 
-//This covers testing vehicle stability and finding a safe position
-private _emptyPosAGL = [_vehicle, _item, _unloader] call EFUNC(common,findUnloadPosition);
-TRACE_1("findUnloadPosition",_emptyPosAGL);
-
-if ((count _emptyPosAGL) != 3) exitWith {
-    TRACE_4("Could not find unload pos",_vehicle,getPosASL _vehicle,isTouchingGround _vehicle,speed _vehicle);
-    if ((!isNull _unloader) && {_unloader == ACE_player}) then {
-        //display text saying there are no safe places to exit the vehicle
-        [localize ELSTRING(common,NoRoomToUnload)] call EFUNC(common,displayTextStructured);
-    };
-    false
+// Get config sensitive case name
+if (_item isEqualType "") then {
+    _item = _item call EFUNC(common,getConfigName);
 };
 
+// Check if item is actually part of cargo
 private _loaded = _vehicle getVariable [QGVAR(loaded), []];
 
 if !(_item in _loaded) exitWith {
-    ERROR_3("Tried to unload item [%1] not in vehicle[%2] cargo[%3]", _item, _vehicle, _loaded);
-    false
+    ERROR_3("Tried to unload item [%1] not in vehicle[%2] cargo[%3]",_item,_vehicle,_loaded);
+
+    false // return
+};
+
+// Check if item can be unloaded
+private _itemSize = _item call FUNC(getSizeItem);
+
+if (_itemSize < 0) exitWith {
+    false // return
+};
+
+// This covers testing vehicle stability and finding a safe position
+private _emptyPosAGL = [_vehicle, _item, _unloader] call EFUNC(common,findUnloadPosition);
+TRACE_1("findUnloadPosition",_emptyPosAGL);
+
+if (_emptyPosAGL isEqualTo []) exitWith {
+    // Display text saying there are no safe places to exit the vehicle
+    if (!isNull _unloader && {_unloader == ACE_player}) then {
+        [ELSTRING(common,NoRoomToUnload)] call EFUNC(common,displayTextStructured);
+    };
+
+    false // return
 };
 
+// Unload item from cargo
 _loaded deleteAt (_loaded find _item);
 _vehicle setVariable [QGVAR(loaded), _loaded, true];
 
-private _space = [_vehicle] call FUNC(getCargoSpaceLeft);
-private _itemSize = [_item] call FUNC(getSizeItem);
-_vehicle setVariable [QGVAR(space), (_space + _itemSize), true];
+// Update cargo space remaining
+private _cargoSpace = _vehicle call FUNC(getCargoSpaceLeft);
+_vehicle setVariable [QGVAR(space), _cargoSpace + _itemSize, true];
 
 private _object = _item;
+
 if (_object isEqualType objNull) then {
     detach _object;
+
     // hideObjectGlobal must be executed before setPos to ensure light objects are rendered correctly
-    // do both on server to ensure they are executed in the correct order
+    // Do both on server to ensure they are executed in the correct order
     [QGVAR(serverUnload), [_object, _emptyPosAGL]] call CBA_fnc_serverEvent;
 } else {
     _object = createVehicle [_item, _emptyPosAGL, [], 0, "NONE"];
```

Comment:
Not the point of the PR, but unloading always being handled on the server makes more sense IMO, instead of this.

Reply:
If you want to add a comment, what would it be? I'm not sure if such a comment is necessary.

Reply:
I'm referring to line 74 and below, virtual items being created locally and then unloaded vs existing items being unloaded on the server and then passed along

Reply:
So you would want the server to handle it fully? So decide if it needs to be actually unloaded or created?

Reply:
Yes, would also make it easier to make everything event-based for the menu instead of running a PFH

Reply:
It's not easily possible, given the current structure, seeing as we need the reference to the object after it's been created.

I guess we could split `FUNC(unloadItem)` and `FUNC(paradropItem)` in half, so it would result in `FUNC(unloadItem)`, `FUNC(paradropItem)` and `FUNC(paradropItemServer)`, where `FUNC(unloadItem)` and FUNC(paradropItem) would remove the server-sided code.

Reply:
IMO should be done in a separate PR to limit scope

Reply:
> IMO should be done in a separate PR to limit scope

That's just what I was thinking.